wavefront/server/modules/agents_module/agents_module/controllers/async_inference_controller.py

```
@@ -0,0 +1,229 @@
+            }
+        ),
+    )
+
+
+@async_router.get('/v1/agentic-executions/{execution_id}')
+@inject
+async def get_execution_status(
+    execution_id: UUID,
+    async_agentic_execution_service: AsyncAgenticExecutionService = Depends(
+        Provide[AgentsContainer.async_agentic_execution_service]
+    ),
```

Ruff (0.15.12)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)